src/js/components/Audio/Audio.js

```diff
@@ -0,0 +1,276 @@
+    mediaEvents.reduce((previousValue, currentValue) => {
+      const nextValue = { ...previousValue };
+      nextValue[currentValue] = e => {
+        if (
+          currentValue in this.props &&
+          /* eslint-disable react/destructuring-assignment */
+          typeof this.props[currentValue] === 'function'
+        ) {
+          this.props[currentValue](e);
+          /* eslint-enable react/destructuring-assignment */
+        }
+        this.update();
+      };
+
+      return nextValue;
+    }, {});
+
+  update = () => {
+    const { audioRef } = this.state;
+    const audio = audioRef.current;
+
+    let { interacting } = this.state;
+    if (audio && audio.ended) {
+      interacting = false;
+    }
+
+    let scrubberUpdate = 0;
+    if (audio.duration) {
+      // smooth scrubbing - total number of segments to use for the scrubber are equal to the duration in milliseconds
+      // the step is reciprocal to the number of segments for smooth transition
+      this.scrubberStep = 1 / (audio.duration * 1000);
+      scrubberUpdate = audio.currentTime / audio.duration;
+    }
+
+    this.setState({
+      currentTime: audio.currentTime,
+      duration: audio.duration,
+      interacting,
+      playing: !audio.paused,
+      scrubberValue: scrubberUpdate,
+    });
+  };
+
+  play = () => {
+    const { audioRef } = this.state;
+    audioRef.current.play();
+  };
+
+  pause = () => {
+    const { audioRef } = this.state;
+    audioRef.current.pause();
+  };
+
+  interactionStart = () => {
+    this.setState({ interacting: true });
+    clearTimeout(this.interactionTimer);
+    this.interactionTimer = setTimeout(this.interactionStop, 3000);
+  };
+
+  interactionStop = () => {
+    const { focus } = this.state;
+    if (!focus && !this.unmounted) {
+      this.setState({ interacting: false });
+    }
+  };
+
+  setVolume = value => {
+    const { audioRef } = this.state;
+    audioRef.current.volume = value;
+    this.setState({ volumeValue: value });
+  };
+
+  seek = value => {
+    const { audioRef, duration } = this.state;
+
+    if (audioRef.current) {
+      const precentageProgress = value * duration;
+      audioRef.current.currentTime = precentageProgress;
+      this.setState({ scrubberValue: value });
+    }
+  };
+
+  renderControls() {
+    const { theme } = this.props;
+    const {
+      currentTime,
+      duration,
+      interacting,
+      playing,
+      scrubberValue,
+      volumeValue,
+    } = this.state;
+
+    const background = (theme.audio.controls &&
+      theme.audio.controls.background) || {
+      color: 'dark-1',
+      opacity: 'strong',
+    };
+
+    const durationFormattedTime = formatTime(duration);
+    const scrubFormattedTime = formatTime(currentTime);
+
+    const Icons = {
+      Pause: theme.audio.icons.pause,
+      Play: theme.audio.icons.play,
+      Volume: theme.audio.icons.volume,
+    };
+
+    return (
+      <StyledAudioControls active={interacting}>
```

Comment:
Why both `StyledAudioControls` and an outer `Box`? I would think these could be merged.